webapp/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { Box, useTheme } from '@mui/material';
+import GlobalStyles from '@mui/material/GlobalStyles';
+import { FocusTrap } from '@mui/base/FocusTrap';
+import Paper from '@mui/material/Paper';
+import {
+  FC,
+  ReactNode,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+} from 'react';
+
+import SidebarOverlay from '@/components/SidebarOverlay';
+import { SidebarContext } from './SidebarContext';
+
+type SidebarProps = {
+  children: ReactNode;
+};
+
+const Sidebar: FC<SidebarProps> = ({ children }) => {
+  const theme = useTheme();
+  const { isSidebarOpen, setIsSidebarOpen } = useContext(SidebarContext);
+
+  const openOnMobile = useMemo(
+    () =>
+      isSidebarOpen &&
+      typeof window !== 'undefined' &&
+      window.innerWidth < theme.breakpoints.values.md,
+    [isSidebarOpen, theme.breakpoints.values.md],
+  );
+
+  const onKeypress = useCallback(
+    (event: KeyboardEvent) => {
+      if (openOnMobile && event.key === 'Escape') {
+        setIsSidebarOpen(false);
+      }
+    },
+    [openOnMobile, setIsSidebarOpen],
+  );
+
+  useEffect(() => {
+    document.addEventListener('keydown', onKeypress);
+    return () => document.removeEventListener('keydown', onKeypress);
+  }, [onKeypress]);
+
+  return (
+    <FocusTrap open={openOnMobile}>
+      <Paper
+        sx={{
+          borderColor: 'divider',
+          borderRight: '1px solid',
+          display: 'flex',
+          flexDirection: 'column',
+          flexShrink: 0,
+          gap: 2,
+          height: '100dvh',
+          position: { md: 'sticky', xs: 'fixed' },
+          top: 0,
+          transform: {
+            md: 'none',
+            xs: 'translateX(calc(100% * (var(--SideNavigation-slideIn, 0) - 1)))',
+          },
+          transition: 'transform 0.4s, width 0.4s',
+          width: 'var(--Sidebar-width)',
+          zIndex: 10000,
+        }}
+      >
```

Comment:
On 3b6fda010a7e5d0cd0b39999955e4d59984b15b9, "MUI: The modal content node does not accept focus." is logged as an error to the console every time I open the sidebar. Looking at the stack trace, it looks like this is logged when the child element has no `tabIndex`.

That's a little harsh of MUI to log an error just for that. We can silence it by adding `tabIndex={-1}` to our `Paper`, or we can open an issue with MUI, but we can also just live with the console error.

Reply:
Excellent notes here, this will be very easy to tidy up.

Reply:
https://github.com/zetkin/lyra/pull/116